### PR TITLE
Fix can't get todo item when it is created for the first time in UI

### LINKED ISSUE #3959: TodoItems does not have Id when created in the TodoTemplate project
When we create a new to-do item in the TodoTemplate project, they don't have an id property, and when we select the checkbox for select that to-do item, it returns an error, and it says that we don't have an id for that to-do item.  
It doesn't been fixed until the refresh page.

### PR DESCRIPTION
**Pull request by hootanht**
this closes #3959

### CODE REVIEW

File: src/Templates/TodoTemplate/Bit.TodoTemplate/src/Client/Shared/Pages/TodoPage.razor.cs [hunk 1]
```diff
@@ -53,8 +53,7 @@ private void FilterViewTodoItems()
     {
         _viewTodoItems = _allTodoItems.Where(todo =>
             _selectedFilter == nameof(AppStrings.Active) ? todo.IsDone == false
-            : _selectedFilter == nameof(AppStrings.Completed) ? todo.IsDone
-            : true
+            : _selectedFilter != nameof(AppStrings.Completed) || todo.IsDone
```
Comment (msynk) [inline src/Templates/TodoTemplate/Bit.TodoTemplate/src/Client/Shared/Pages/TodoPage.razor.cs:56]: please undo this change, it is not related to this issue and the old version is easier to understand.

File: src/Templates/TodoTemplate/Bit.TodoTemplate/src/Client/Shared/Pages/TodoPage.razor.cs [hunk 2]
```diff
@@ -115,7 +114,11 @@ private async Task AddTodoItem()
         {
             var newTodoItem = new TodoItemDto { Title = _newTodoTitle, Date = DateTimeOffset.Now };
 
-            await HttpClient.PostAsJsonAsync("TodoItem/Create", newTodoItem, AppJsonContext.Default.TodoItemDto);
+            var response = await HttpClient.PostAsJsonAsync("TodoItem/Create", newTodoItem, AppJsonContext.Default.TodoItemDto);
+
+            var content = await response.Content.ReadAsStringAsync();
+
+            newTodoItem.Id = int.Parse(content);
```
Comment (msynk) [inline src/Templates/TodoTemplate/Bit.TodoTemplate/src/Client/Shared/Pages/TodoPage.razor.cs:121]: plz change these lines to this:

```csharp
using var response = await HttpClient.PostAsJsonAsync("TodoItem/Create", newTodoItem, AppJsonContext.Default.TodoItemDto);
newTodoItem.Id = await response.Content.ReadFromJsonAsync<int>();
```

File: src/Templates/TodoTemplate/Bit.TodoTemplate/src/Server/Api/appsettings.json [hunk 3]
```diff
@@ -30,7 +30,7 @@
         "EmailSettings": {
             "Host": "LocalFolder", // Local folder means storing emails in bin\sent-emails folder (Recommended for testing purposes only) instead of sending them using smtp server.
             "Port": "25",
-            "DefaulFromEmail": "info@TodoTemplate.com",
+            "DefaultFromEmail": "info@TodoTemplate.com",
```
Comment (msynk) [inline src/Templates/TodoTemplate/Bit.TodoTemplate/src/Server/Api/appsettings.json:33]: this is not also related to this issue

**Review by msynk — CHANGES_REQUESTED** [review]

File: src/Templates/TodoTemplate/Bit.TodoTemplate/src/Client/Shared/Pages/TodoPage.razor.cs (hunk 1 above)
Comment (hootanht) [reply src/Templates/TodoTemplate/Bit.TodoTemplate/src/Client/Shared/Pages/TodoPage.razor.cs]: Done

File: src/Templates/TodoTemplate/Bit.TodoTemplate/src/Client/Shared/Pages/TodoPage.razor.cs (hunk 2 above)
Comment (hootanht) [reply src/Templates/TodoTemplate/Bit.TodoTemplate/src/Client/Shared/Pages/TodoPage.razor.cs]: Done

File: src/Templates/TodoTemplate/Bit.TodoTemplate/src/Server/Api/appsettings.json (hunk 3 above)
Comment (hootanht) [reply src/Templates/TodoTemplate/Bit.TodoTemplate/src/Server/Api/appsettings.json]: Done

**Review by msynk — APPROVED** [review]
